epochStart/shardchain/peerMiniBlocksSyncer_test.go

```diff
@@ -344,3 +345,458 @@ func TestValidatorInfoProcessor_ProcesStartOfEpochWithMissinPeerMiniblocksTimeou
+
+		assert.Nil(t, err)
+		assert.Nil(t, missingValidatorsInfoHashes)
+		assert.Equal(t, 3, len(validatorsInfo))
+		assert.Equal(t, svi1, validatorsInfo["a"])
+		assert.Equal(t, svi2, validatorsInfo["b"])
+		assert.Equal(t, svi3, validatorsInfo["c"])
+	})
+}
+
+func TestValidatorInfoProcessor_ReceivedValidatorInfo(t *testing.T) {
+	t.Parallel()
+
+	t.Run("received validators info with wrong type assertion", func(t *testing.T) {
+		t.Parallel()
+
+		args := createDefaultArguments()
+		syncer, _ := NewPeerMiniBlockSyncer(args)
+		syncer.initValidatorsInfo()
+
+		syncer.mutValidatorsInfoForBlock.Lock()
+		syncer.mapAllValidatorsInfo["a"] = nil
+		syncer.numMissingValidatorsInfo = 1
+		syncer.mutValidatorsInfoForBlock.Unlock()
+
+		syncer.receivedValidatorInfo([]byte("a"), nil)
+
+		syncer.mutValidatorsInfoForBlock.RLock()
+		numMissingValidatorsInfo := syncer.numMissingValidatorsInfo
+		syncer.mutValidatorsInfoForBlock.RUnlock()
+
+		assert.Equal(t, uint32(1), numMissingValidatorsInfo)
+	})
+
+	t.Run("received validators info with not requested validator info", func(t *testing.T) {
+		t.Parallel()
+
+		svi := &state.ShardValidatorInfo{PublicKey: []byte("x")}
+
+		args := createDefaultArguments()
+		syncer, _ := NewPeerMiniBlockSyncer(args)
+		syncer.initValidatorsInfo()
+
+		syncer.mutValidatorsInfoForBlock.Lock()
+		syncer.mapAllValidatorsInfo["a"] = nil
+		syncer.numMissingValidatorsInfo = 1
+		syncer.mutValidatorsInfoForBlock.Unlock()
+
+		syncer.receivedValidatorInfo([]byte("b"), svi)
+
+		syncer.mutValidatorsInfoForBlock.RLock()
+		numMissingValidatorsInfo := syncer.numMissingValidatorsInfo
+		syncer.mutValidatorsInfoForBlock.RUnlock()
+
+		assert.Equal(t, uint32(1), numMissingValidatorsInfo)
+	})
+
+	t.Run("received validators info with already received validator info", func(t *testing.T) {
+		t.Parallel()
+
+		svi := &state.ShardValidatorInfo{PublicKey: []byte("x")}
+
+		args := createDefaultArguments()
+		syncer, _ := NewPeerMiniBlockSyncer(args)
+		syncer.initValidatorsInfo()
+
+		syncer.mutValidatorsInfoForBlock.Lock()
+		syncer.mapAllValidatorsInfo["a"] = svi
+		syncer.numMissingValidatorsInfo = 1
+		syncer.mutValidatorsInfoForBlock.Unlock()
+
+		syncer.receivedValidatorInfo([]byte("a"), svi)
+
+		syncer.mutValidatorsInfoForBlock.RLock()
+		numMissingValidatorsInfo := syncer.numMissingValidatorsInfo
+		syncer.mutValidatorsInfoForBlock.RUnlock()
+
+		assert.Equal(t, uint32(1), numMissingValidatorsInfo)
+	})
+
+	t.Run("received validators info with missing validator info", func(t *testing.T) {
+		t.Parallel()
+
+		svi := &state.ShardValidatorInfo{PublicKey: []byte("x")}
+
+		args := createDefaultArguments()
+		syncer, _ := NewPeerMiniBlockSyncer(args)
+		syncer.initValidatorsInfo()
+
+		syncer.mutValidatorsInfoForBlock.Lock()
+		syncer.mapAllValidatorsInfo["a"] = nil
+		syncer.numMissingValidatorsInfo = 1
+		syncer.mutValidatorsInfoForBlock.Unlock()
+
+		var err error
```

Comment:
please use a mutex to protect the read or write of the err error. Reason: you write it asynchronously on the go routine

Reply:
Refactored wiith atomic.Flag